runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/graph/CreateExecutableStageNodeFunction.java

```diff
@@ -260,11 +249,13 @@ public Node apply(MutableNetwork<Node, Edge> input) {
             e);
       }
 
+      // TODO(BEAM-6275): Set correct IsBounded on generated PCollections
```

Comment:
The function here is bonkers long. Not your responsibility but if your hacking lent you any insight, please do split it into pieces like 1/10 the size.

Reply:
I can follow-up in a subsequent PR